presto-main/src/main/java/com/facebook/presto/operator/HashAggregationOperator.java

```diff
@@ -360,20 +568,159 @@ protected Page computeNext()
                             BlockBuilder output = pageBuilder.getBlockBuilder(types.size() + i);
                             aggregator.evaluate(groupId, output);
                         }
-
-                        groupId++;
                     }
 
                     return pageBuilder.build();
                 }
             };
         }
+
+        private void rebuildAggregators()
```

Comment:
having just partial/final differentiation is easier since there is just one naming dimension. When you also add merge/not merge differentiation it makes things harder to understand. One has to know that merge is final/not merge is partial. When you call `rebuildAggregatorsForMerge` all you really do is `rebuildPartialAggregators`